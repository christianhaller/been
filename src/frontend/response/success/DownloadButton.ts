import bytes from "bytes";

export class DownloadButton {
  private el: HTMLAnchorElement;

  init(parent: HTMLElement, blob: Blob, username: string) {
    this.el = parent.querySelector("#download");

    this.el.href = window.URL.createObjectURL(blob);
    this.el.download = `${username}.zip`;

    this.el.classList.remove("cursor-not-allowed");
    this.el.querySelector("span").innerText = `Download (${bytes(blob.size)})`;
  }
}
