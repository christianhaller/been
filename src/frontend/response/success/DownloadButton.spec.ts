import { DownloadButton } from "./DownloadButton";

describe("DownloadButton", () => {
  document.body.innerHTML =
    '<a id="download" class="cursor-not-allowed" ><span></span></a>';

  global.URL.createObjectURL = jest.fn(() => {
    return "x";
  });

  const sut = new DownloadButton();
  test("init", () => {
    sut.init(document.body, new Blob(), "christian");
    const button = document.querySelector("#download");
    expect(button.classList.contains("cursor-not-allowed")).toBe(false);
    expect(button).toMatchSnapshot();
  });
});
