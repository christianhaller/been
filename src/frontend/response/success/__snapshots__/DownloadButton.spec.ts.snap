// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton init 1`] = `
<a
  class="px-4 py-2 border-transparent text-base leading-6 font-medium rounded-md text-white bg-indigo-600 inline-flex items-center hover:bg-indigo-500 focus:outline-none focus:border-indigo-700 focus:shadow-outline-indigo active:bg-indigo-700"
  download="christian.zip"
  href="x"
  id="download"
>
  <svg
    class="animate-bounce fill-current w-4 h-4 mr-2"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
    />
  </svg>
  <span>
    Download
  </span>
</a>
`;
